src/Microsoft.DotNet.ProjectModel/ProjectSystem/ProjectSystemCache.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.DotNet.ProjectModel.Caching;
+using NuGet.Frameworks;
+
+namespace Microsoft.Extensions.ProjectModel.ProjectSystem
+{
+    internal class ProjectSystemCache
```

Comment:
This can be simplified. Get rid of all the things that we don't need in the short term and get rid of the Task.Run calls.